sentry_sdk/integrations/falcon.py

```diff
@@ -58,16 +73,27 @@ def raw_data(self):
         else:
             return None
 
-    def json(self):
-        # type: () -> Optional[Dict[str, Any]]
-        try:
-            return self.request.media
-        except falcon.errors.HTTPBadRequest:
-            # NOTE(jmagnusson): We return `falcon.Request._media` here because
-            # falcon 1.4 doesn't do proper type checking in
-            # `falcon.Request.media`. This has been fixed in 2.0.
-            # Relevant code: https://github.com/falconry/falcon/blob/1.4.1/falcon/request.py#L953
-            return self.request._media
+    if FALCON3:
+
+        def json(self):
```

Comment:
I'd just merge this into two `hasattr` checks for `media/_media` and pick whichever is available

Reply:
Falcon 3 and Falcon 2 `Request` classes have `_media` attribute and `media` property. How would this `hasattr` check look like?

Reply:
I guess this is then fine as it is. It is not the most elegant code, but it should work.